lightcycle-lib/src/main/java/com/soundcloud/lightcycle/FragmentLightCycleDispatcher.java

```diff
@@ -32,20 +34,27 @@ public void bind(FragmentLightCycle<T> lightCycle) {
 
     @Override
     public void onAttach(T fragment, Activity activity) {
-        LightCycles.bind(this);
+        bindIfNecessary();
         for (FragmentLightCycle<T> component : fragmentLightCycles) {
             component.onAttach(fragment, activity);
         }
     }
 
     @Override
     public void onAttach(T fragment, Context context) {
-        LightCycles.bind(this);
+        bindIfNecessary();
         for (FragmentLightCycle<T> component : fragmentLightCycles) {
             component.onAttach(fragment, context);
         }
     }
 
+    private void bindIfNecessary() {
```

Comment:
I heard and actually added an integration test. 

Because we use static calls and reflexion to bind lightcycles, I did not know how to test it at the unit level.